src/core/symbology/qgssvgcache.cpp

```diff
@@ -406,77 +409,75 @@ QByteArray QgsSvgCache::getImageData( const QString &path ) const
-    while ( !reply->isFinished() )
+    QNetworkReply *reply = task->reply();
+    if ( !reply )
     {
-      QCoreApplication::processEvents( QEventLoop::ExcludeUserInputEvents, 500 );
+      // canceled
+      QMetaObject::invokeMethod( const_cast< QgsSvgCache * >( this ), "onRemoteSvgFetched", Qt::QueuedConnection, Q_ARG( QString, path ), Q_ARG( bool, false ) );
+      return;
     }
 
     if ( reply->error() != QNetworkReply::NoError )
     {
       QgsMessageLog::logMessage( tr( "SVG request failed [error: %1 - url: %2]" ).arg( reply->errorString(), path ), tr( "SVG" ) );
-      reply->deleteLater();
-      return mMissingSvg;
+      return;
     }
 
-    QVariant redirect = reply->attribute( QNetworkRequest::RedirectionTargetAttribute );
-    if ( redirect.isNull() )
+    bool ok = true;
+
+    QVariant status = reply->attribute( QNetworkRequest::HttpStatusCodeAttribute );
+    if ( !status.isNull() && status.toInt() >= 400 )
     {
-      // neither network error nor redirection
-      // TODO: cache the image
-      break;
+      QVariant phrase = reply->attribute( QNetworkRequest::HttpReasonPhraseAttribute );
+      QgsMessageLog::logMessage( tr( "SVG request error [status: %1 - reason phrase: %2] for %3" ).arg( status.toInt() ).arg( phrase.toString(), path ), tr( "SVG" ) );
+      mRemoteContentCache.insert( path, new QByteArray( mMissingSvg ) );
+      ok = false;
     }
 
-    // do a new request to the redirect url
-    svgUrl = redirect.toUrl();
-    reply->deleteLater();
-  }
-
-  QVariant status = reply->attribute( QNetworkRequest::HttpStatusCodeAttribute );
-  if ( !status.isNull() && status.toInt() >= 400 )
-  {
-    QVariant phrase = reply->attribute( QNetworkRequest::HttpReasonPhraseAttribute );
-    QgsMessageLog::logMessage( tr( "SVG request error [status: %1 - reason phrase: %2] for %3" ).arg( status.toInt() ).arg( phrase.toString(), path ), tr( "SVG" ) );
-
-    reply->deleteLater();
-    return mMissingSvg;
-  }
-
-  // we accept both real SVG mime types AND plain text types - because some sites
-  // (notably github) serve up svgs as raw text
-  QString contentType = reply->header( QNetworkRequest::ContentTypeHeader ).toString();
-  if ( !contentType.startsWith( QLatin1String( "image/svg+xml" ), Qt::CaseInsensitive )
-       && !contentType.startsWith( QLatin1String( "text/plain" ), Qt::CaseInsensitive ) )
-  {
-    QgsMessageLog::logMessage( tr( "Unexpected MIME type %1 received for %2" ).arg( contentType, path ), tr( "SVG" ) );
-    reply->deleteLater();
-    return mMissingSvg;
-  }
+    // we accept both real SVG mime types AND plain text types - because some sites
+    // (notably github) serve up svgs as raw text
+    QString contentType = reply->header( QNetworkRequest::ContentTypeHeader ).toString();
+    if ( !contentType.startsWith( QLatin1String( "image/svg+xml" ), Qt::CaseInsensitive )
+         && !contentType.startsWith( QLatin1String( "text/plain" ), Qt::CaseInsensitive ) )
```

Comment:
What are we protecting against with the mime-type check?

Reply:
I assume it was originally put there to protect against trying to load a png or other binary into the svg renderer